Add support for file's builtin magic module

felixonmars opened the pull request:
file's builtin magic module has a somewhat different API, and it's not
co-installable with pypi:python-magic as both has the same name. Adding
a fallback logic here makes mocket work with either one.

All related tests are passing here.

mindflayer commented:
Hi @felixonmars, my main concern is about how to test the two solutions, since they have the same namespace `import magic`. Fixing the issue you opened (#119) does not seem the most difficult part of the change.

mindflayer commented:
I was having a look at https://pypi.org/project/filemagic/ and I thought it was the module you are trying to support but I see it implements a slightly different API. Is that what you used?

felixonmars commented:
> I was having a look at https://pypi.org/project/filemagic/ and I thought it was the module you are trying to support but I see it implements a slightly different API. Is that what you used?

Actually, it's yet another. The one I use is bundled inside upstream "file" project and unfortunately not available from PyPI: https://github.com/archlinux/svntogit-community/blob/packages/python-magic/trunk/PKGBUILD#L12

So projects like s3cmd has implemented a quite complex compatibility check for the three versions:

https://github.com/s3tools/s3cmd/blob/3d4a4caf91582c2a674deaa837a5e07d96510059/S3/S3.py#L55

mindflayer commented:
> The one I use is bundled inside upstream "file" project and unfortunately not available from PyPI
So the question is: how to test it in a CI environment if not available in PyPI?

felixonmars commented:
>  So the question is: how to test it in a CI environment if not available in PyPI?

I see no easy answer either. Since the Arch repository version is the "file" one, and I always run tests during packaging, would you feel okay if I report back in a timely manner if the tests are not passing?

mindflayer commented:
In that case I will have to exclude that block from `mocket`'s test coverage. Could you try to see if the other `flag` I suggested is available in the library you are testing?

felixonmars commented:
@mindflayer Indeed, that's the problem. I have updated this PR to use MAGIC_MIME_TYPE instead and reverted the test change, all tests are now passing.

mindflayer commented:
OK, let me finish what I am doing for my job and I'll release a version of `mocket` with the changes you need for make it work on Arch.
By the way, I used to be an Arch user, so I am really happy to help.

felixonmars commented:
Thanks a lot :)

mindflayer commented:
@felixonmars Here you are: https://pypi.org/project/mocket/3.8.7/

felixonmars commented:
Thanks!
